mono/metadata/class-init.c

```diff
@@ -3846,40 +3844,35 @@ mono_class_layout_fields (MonoClass *klass, int base_instance_size, int packing_
 			real_size = MAX (real_size, size + field_offsets [i]);
 		}
 
-		if (klass->has_references) {
-			ref_bitmap = g_new0 (guint8, real_size / TARGET_SIZEOF_VOID_P);
-
-			/* Check for overlapping reference and non-reference fields */
-			for (i = 0; i < top; i++) {
-				MonoType *ftype;
-
+		/* check for incorrectly aligned or overlapped by a non-object field */
+#ifdef ENABLE_NETCORE	
```

Comment:
Is there a reason this fix is netcore-only? I suspect the same issue is present on legacy if we tried an equivalent test.

Reply:
Oh, just read PR discussion.